generatebundlefile/ecr_test.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestPullHelmChart(t *testing.T) {
+	tests := []struct {
+		testName        string
+		testHelmName    string
+		testHelmVersion string
+		helmLocation    string
+		wantErr         bool
+	}{
+		{
+			testName:        "Test empty name",
+			testHelmName:    "",
+			testHelmVersion: "0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78",
+			helmLocation:    "",
+			wantErr:         true,
+		},
+		{
+			testName:        "Test empty version",
+			testHelmName:    "646717423341.dkr.ecr.us-west-2.amazonaws.com/hello-eks-anywhere",
+			testHelmVersion: "",
+			helmLocation:    "",
+			wantErr:         true,
+		},
+		{
+			testName:        "Test valid helm",
+			testHelmName:    "646717423341.dkr.ecr.us-west-2.amazonaws.com/hello-eks-anywhere",
+			testHelmVersion: "0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78",
+			helmLocation:    fmt.Sprintf("%s/Library/Caches/helm/repository/hello-eks-anywhere-0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78.tgz", os.Getenv("HOME")),
+			wantErr:         false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			got, err := PullHelmChart(tc.testHelmName, tc.testHelmVersion)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("PullHelmChart() error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if got != tc.helmLocation {
+				t.Fatalf("PullHelmChart() = %#v\n\n\n, want %#v", got, tc.helmLocation)
```

Comment:
```suggestion
				tt.Fatalf("PullHelmChart() error = %v, wantErr %v", err, tc.wantErr)
			}
			if got != tc.helmLocation {
				tt.Fatalf("PullHelmChart() = %#v\n\n\n, want %#v", got, tc.helmLocation)
```
You don't want to lose any of the context that is gained in `tt`.

Reply:
fixed